### PR TITLE
Adds Illuminate's Gen9 effect

### LINKED ISSUE #3478: Illuminate is missing its Gen9 effect
### Description

[Illuminate received an in-battle effect in patch 2.0.1 of SV.](https://bulbapedia.bulbagarden.net/wiki/Illuminate_(Ability)) This effect is still missing from RHH.

### Version

upcoming (Edge)

### Upcoming Version

_No response_

### Discord contact info

bassoonian

### PR DESCRIPTION
**Pull request by Bassoonian**
See title

## Issue(s) that this PR fixes
Fixes #3478 

## **Discord contact info**
bassoonian

### CODE REVIEW

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -870,6 +870,9 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                     RETURN_SCORE_MINUS(10);
                 break;
             case ABILITY_KEEN_EYE:
+        #if B_ILLUMINATE_EFFECT == GEN_9
```
Comment (AsparagusEduardo) [inline src/battle_ai_main.c:873]: We're moving away from #ifdef blocks. Please use proper ifs.

**Comment by Bassoonian** [comment]
Ready for actual review!

**Comment by AlexOn1ine** [comment]
Could you write some tests for Illuminate?

File: src/battle_ai_main.c [hunk 2]
```diff
@@ -1292,7 +1296,7 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
         case EFFECT_ACCURACY_DOWN_2:
             if (!ShouldLowerStat(battlerDef, aiData->abilities[battlerDef], STAT_ACC))
                 ADJUST_SCORE(-10);
-            else if (aiData->abilities[battlerDef] == ABILITY_KEEN_EYE)
+            else if (aiData->abilities[battlerDef] == ABILITY_KEEN_EYE || (aiData->abilities[battlerDef] == ABILITY_ILLUMINATE && B_ILLUMINATE_EFFECT == GEN_9))
```
Comment (AsparagusEduardo) [inline src/battle_ai_main.c:1299]:
```suggestion
            else if (aiData->abilities[battlerDef] == ABILITY_KEEN_EYE || (B_ILLUMINATE_EFFECT >= GEN_9 && aiData->abilities[battlerDef] == ABILITY_ILLUMINATE))
```

File: src/battle_ai_switch_items.c [hunk 3]
```diff
@@ -347,6 +347,7 @@ static bool8 ShouldSwitchIfGameStatePrompt(u32 battler)
             if (gBattleMons[battler].statStages[STAT_EVASION] > (DEFAULT_STAT_STAGE + 3)
                 && AI_DATA->abilities[opposingBattler] != ABILITY_UNAWARE
                 && AI_DATA->abilities[opposingBattler] != ABILITY_KEEN_EYE
+                && (AI_DATA->abilities[opposingBattler] != ABILITY_ILLUMINATE && B_ILLUMINATE_EFFECT == GEN_9)
```
Comment (AsparagusEduardo) [inline src/battle_ai_switch_items.c:350]:
```suggestion
                && (B_ILLUMINATE_EFFECT >= GEN_9 && AI_DATA->abilities[opposingBattler] != ABILITY_ILLUMINATE)
```

File: src/battle_ai_util.c [hunk 4]
```diff
@@ -1996,6 +1996,7 @@ bool32 ShouldLowerAccuracy(u32 battlerAtk, u32 battlerDef, u32 defAbility)
       && defAbility != ABILITY_WHITE_SMOKE
       && defAbility != ABILITY_FULL_METAL_BODY
       && defAbility != ABILITY_KEEN_EYE
+      && (defAbility != ABILITY_ILLUMINATE && B_ILLUMINATE_EFFECT == GEN_9)
```
Comment (AsparagusEduardo) [inline src/battle_ai_util.c:1999]:
```suggestion
      && (B_ILLUMINATE_EFFECT >= GEN_9 && defAbility != ABILITY_ILLUMINATE)
```

File: src/battle_script_commands.c [hunk 5]
```diff
@@ -1623,7 +1623,7 @@ u32 GetTotalAccuracy(u32 battlerAtk, u32 battlerDef, u32 move, u32 atkAbility, u
     gPotentialItemEffectBattler = battlerDef;
     accStage = gBattleMons[battlerAtk].statStages[STAT_ACC];
     evasionStage = gBattleMons[battlerDef].statStages[STAT_EVASION];
-    if (atkAbility == ABILITY_UNAWARE || atkAbility == ABILITY_KEEN_EYE)
+    if (atkAbility == ABILITY_UNAWARE || atkAbility == ABILITY_KEEN_EYE || (atkAbility == ABILITY_ILLUMINATE && B_ILLUMINATE_EFFECT == GEN_9))
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:1626]:
```suggestion
    if (atkAbility == ABILITY_UNAWARE || atkAbility == ABILITY_KEEN_EYE || (B_ILLUMINATE_EFFECT >= GEN_9 && atkAbility == ABILITY_ILLUMINATE))
```

File: src/battle_script_commands.c [hunk 6]
```diff
@@ -5058,6 +5058,7 @@ static void Cmd_playstatchangeanimation(void)
                         && ability != ABILITY_FULL_METAL_BODY
                         && ability != ABILITY_WHITE_SMOKE
                         && !(ability == ABILITY_KEEN_EYE && currStat == STAT_ACC)
+                        && !(ability == ABILITY_ILLUMINATE && currStat == STAT_ACC && B_ILLUMINATE_EFFECT == GEN_9)
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:5061]:
```suggestion
                        && !(B_ILLUMINATE_EFFECT >= GEN_9 && ability == ABILITY_ILLUMINATE && currStat == STAT_ACC)
```

File: src/battle_script_commands.c [hunk 7]
```diff
@@ -11437,6 +11438,7 @@ static u32 ChangeStatBuffs(s8 statValue, u32 statId, u32 flags, const u8 *BS_ptr
         }
         else if (!certain
                 && ((battlerAbility == ABILITY_KEEN_EYE && statId == STAT_ACC)
+                || (battlerAbility == ABILITY_ILLUMINATE && statId == STAT_ACC && B_ILLUMINATE_EFFECT == GEN_9)
```
Comment (AsparagusEduardo) [inline src/battle_script_commands.c:11441]:
```suggestion
                || (B_ILLUMINATE_EFFECT >= GEN_9 && battlerAbility == ABILITY_ILLUMINATE && statId == STAT_ACC)
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**